packages/url-state-provider/src/index.ts

```diff
@@ -152,16 +165,15 @@ function updateState(stateID, state, options = {}) {
  * changes to the method.
  * @param {boolean} historyOptions.replace - If true it replaces the last state in the window history (default false).
  */
-function updateStateAndHistory(stateID, state, merge, historyOptions) {
-  historyOptions = historyOptions || {}
+function updateStateAndHistory(stateID: string, state: object, merge: boolean, historyOptions: HistoryOptions = {}) {
```

Comment:
Instead of casting you could define the type of newUrl, or what do you think?

```typescript
function updateStateAndHistory(stateID: string, state: object, merge: boolean, historyOptions: HistoryOptions = {}) {
  const newUrl: string = updateState(stateID, state, { merge }); // Ensure newUrl is typed as a string
  const historyState = historyOptions.state || "";
  const historyTitle = historyOptions.title || "";

  if (historyOptions?.replace) {
    window.history.replaceState(historyState, historyTitle, newUrl);
  } else {
    window.history.pushState(historyState, historyTitle, newUrl);
  }
}
```